apps/bare-expo/scripts/build-detox-ios.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+# This script is design to wrap the xcodebuild command
+# and exit with non-zero if the build fails. 
+#
+# This ensures that CI fails on the correct step 
+# instead of attempting to run Detox tests without a build.
+
+set -e
```

Comment:
```suggestion
set -euo pipefail
```

Include `-u` so we catch undefined variables, but also give `-o pipefail` a try. What I expect to happen is that `xcodebuild | tee | xcpretty` will run to completion, but with `-o pipefail` Bash will notice that xcodebuild exited with a non-zero code and then exit early. If this works as expected, we can remove the tee + grep code.